### PR TITLE
Migrate XSD into galaxy-tool-util.

### PR DESCRIPTION
**Pull request by jmchilton**
Builds on #8062, #7971, and #8051.

### CODE REVIEW

**Comment by nsoranzo** [comment]
Need to update:
- [x] `doc/Makefile`
- [x] `scripts/validate_tools.sh`

**Review by mvdbeek — APPROVED** [review]

File: packages/test.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# containers has no tests, tool_util not yet working 100%,
+# data has many problems quota, tool shed install database, etc..
+RUN_TESTS=(1 1 1 0 0 0 0)
+
+for ((i=0; i<${#PACKAGE_DIRS[@]}; i++)); do
+    package_dir=${PACKAGE_DIRS[$i]}
+    run_tests=${RUN_TESTS[$i]}
+
+    cd "$package_dir"
+    pip install -e .
```
Comment (mvdbeek) [inline packages/test.sh:33]: Should we add `--index-url https://wheels.galaxyproject.org/simple --extra-index-url https://pypi.python.org/simple` to get bx-python wheels ?
The galaxy-daya install seems to fail with

```
Building wheels for collected packages: bx-python, SQLAlchemy, tzlocal, bagit, python-openid, Tempita
  Building wheel for bx-python (setup.py) ... - error
  ERROR: Complete output from command /tmp/gxpkgtestenvODtzjt/bin/python -u -c 'import setuptools, tokenize;__file__='"'"'/tmp/pip-install-yIvHjU/bx-python/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-UDdsfA --python-tag cp27:
  ERROR: Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-yIvHjU/bx-python/setup.py", line 223, in <module>
      main()
    File "/tmp/pip-install-yIvHjU/bx-python/setup.py", line 74, in main
      raise Exception( "numpy must be installed to build" )
  Exception: numpy must be installed to build
  ----------------------------------------
  ERROR: Failed building wheel for bx-python
```

Comment (jmchilton) [reply packages/test.sh:33]: I'd really like these libraries to be as vanilla as possible and avoid this Galaxy-ism. Maybe we could publish our bx-python wheels to PyPI instead? I'll see who I can ping on that (https://github.com/bxlab/bx-python/issues/42).

Comment (mvdbeek) [reply packages/test.sh:33]: Clearly the best option.

**Pull request merged**